engine/renderer/private/pipelines/ibl_pipeline.cpp

```diff
@@ -20,30 +32,26 @@ IBLPipeline::IBLPipeline(const VulkanBrain& brain, ResourceHandle<Image> environ
 
 IBLPipeline::~IBLPipeline()
 {
+    auto vkContext { _context->VulkanContext() };
+
     for (const auto& mips : _prefilterMapViews)
         for (const auto& view : mips)
-            _brain.device.destroy(view);
-
-    _brain.GetImageResourceManager().Destroy(_irradianceMap);
-    _brain.GetImageResourceManager().Destroy(_brdfLUT);
-    _brain.GetImageResourceManager().Destroy(_prefilterMap);
-
-    _brain.device.destroy(_sampler);
-
-    _brain.device.destroy(_prefilterPipeline);
-    _brain.device.destroy(_prefilterPipelineLayout);
-    _brain.device.destroy(_irradiancePipeline);
-    _brain.device.destroy(_irradiancePipelineLayout);
-    _brain.device.destroy(_brdfLUTPipeline);
-    _brain.device.destroy(_brdfLUTPipelineLayout);
+            vkContext->Device().destroy(view);
+
+    vkContext->Device().destroy(_prefilterPipeline);
+    vkContext->Device().destroy(_prefilterPipelineLayout);
+    vkContext->Device().destroy(_irradiancePipeline);
+    vkContext->Device().destroy(_irradiancePipelineLayout);
+    vkContext->Device().destroy(_brdfLUTPipeline);
+    vkContext->Device().destroy(_brdfLUTPipelineLayout);
 }
 
 void IBLPipeline::RecordCommands(vk::CommandBuffer commandBuffer)
 {
-    const Image& irradianceMap = *_brain.GetImageResourceManager().Access(_irradianceMap);
-    const Image& prefilterMap = *_brain.GetImageResourceManager().Access(_prefilterMap);
+    const Image& irradianceMap = *_context->Resources()->ImageResourceManager().Access(_irradianceMap);
+    const Image& prefilterMap = *_context->Resources()->ImageResourceManager().Access(_prefilterMap);
 
-    util::BeginLabel(commandBuffer, "Irradiance pass", glm::vec3 { 17.0f, 138.0f, 178.0f } / 255.0f, _brain.dldi);
+    util::BeginLabel(commandBuffer, "Irradiance pass", glm::vec3 { 17.0f, 138.0f, 178.0f } / 255.0f, _context->VulkanContext()->Dldi());
 
     util::TransitionImageLayout(commandBuffer, irradianceMap.image, irradianceMap.format, vk::ImageLayout::eUndefined, vk::ImageLayout::eColorAttachmentOptimal, 6, 0, 1);
```

Comment:
having all of these essential commands inside of a util:: namespace is not as convenient as a dedicated CommandList class. It might be worth considering to create an abstractions that already has easier to use functions for common actions like: Setting pass name, setting barriers, binding descriptors.